cli/src/main/java/hudson/cli/FullDuplexHttpStream.java

```diff
@@ -48,16 +48,31 @@ private static String basicAuth(String userInfo) {
         return null;
     }
 
+    @Deprecated
+    public FullDuplexHttpStream(URL target, String authorization) throws IOException {
+        this(new URL(target.toString().replaceFirst("/cli.*$", "")), target.toString().replaceFirst("(.+)/cli.*$", "$1"), authorization);
```

Comment:
This seems to be being altogether too clever for it's own good. at least have a comment explaining what the URL transformation is supposed to be